### ISSUE TITLE
[2.4.0] Problem when returning arrays within a `case` statement

### ISSUE THREAD

**Issue by fquffio** [squizlabs/PHP_CodeSniffer#804]
There seems to be an issue with indentation in switches, when an array is returned within a `case` statement.

This is the file `MyClass.php` (that, as far as I can tell, is PSR-2 compliant):

```php
<?php

namespace MyApp;

class MyClass
{
    public function test($lang = 'en')
    {
        switch ($lang) {
            case 'en':
                return ['hello', 'world'];
            case 'it':
                return ['ciao', 'mondo'];
        }
    }
}
```

When I run `phpcs --standard=PSR2 MyClass.php` i have this output:

```
FILE: /var/www/myapp/MyClass.php
----------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------
 12 | ERROR | [x] Line indented incorrectly; expected 16 spaces,
    |       |     found 12
----------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------

Time: 17ms; Memory: 4.25Mb
```

The sniffer is OK until the `return ['hello', 'world'];`, then apparently fails to recognize that the "scope" has ended, thus expects the `case 'it':` to be at the same indent level as the return on the previous line.

The same problem seems to happen also with other standards, even if (of course) other warnings pop up.

Also, if the new array syntax (`[]`) is replaced with the legacy `array()`, everything works as expected.

My system:

```
# php -v
PHP 5.6.16 (cli) (built: Dec  2 2015 21:10:10) 
Copyright (c) 1997-2015 The PHP Group
Zend Engine v2.6.0, Copyright (c) 1998-2015 Zend Technologies
    with Xdebug v2.3.3, Copyright (c) 2002-2015, by Derick Rethans

# phpcs --version
PHP_CodeSniffer version 2.4.0 (stable) by Squiz (http://www.squiz.net)
```

**Comment by aik099** [comment]
Please try searching for previously reported issues about same topic (including closed ones) before reporting. In this case this is 5th duplicate of issue, that was already fixed in `master`, but might not be released yet.

**Comment by fquffio** [comment]
My apologies, I probably tried with wrong search terms.

**Comment by aik099** [comment]
Best approach is to try reproducing on `master` to see if problem is still there.
